lib/api/core/notifier.js

```diff
@@ -258,8 +258,42 @@ class NotifierController {
    * @param {Array} ids - list of deleted document IDs
    */
   notifyDocumentDelete(request, ids) {
-    return Bluebird.each(ids, (id, callback) => {
-      return this.kuzzle.services.list.internalCache.search(this.cacheKeyPrefix + id)
+    return Bluebird.all(ids.map(id => {
+      const
+        getRequest = new Request({
```

Comment:
or ignore search from cachedRooms and only get the matchedRooms from DSL (as the `dsl.test` method should be called anyway, fetching some rooms from internalCache before will not improve performances)